### PR TITLE
Saas 16 who said stable yti

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sales_team/views/crm_team_views.xml [hunk 1]
```diff
@@ -18,6 +18,8 @@
             <menuitem action="base.action_partner_category_form" id="menu_partner_category_form" name="Contact Tags" sequence="1" parent="menu_config_address_book" groups="base.group_no_one"/>
             <menuitem action="base.action_partner_title_contact" id="menu_partner_title_contact" name="Contact Titles" parent="menu_config_address_book" sequence="3" groups="base.group_no_one"/>
             <menuitem id="menu_localisation" name="Localization" parent="menu_config_address_book" sequence="5" groups="base.group_no_one"/>
+            <menuitem id="res_partner_industry_menu" name="Sector of Activity" action="base.res_partner_industry_action" parent="sales_team.menu_config_address_book" sequence="7" groups="base.group_no_one"/>
```
Comment (odony) [inline addons/sales_team/views/crm_team_views.xml]: "Sector**s**" should be plural. And perhaps we could move it a bit higher in the sequence order (e.g. 4) to avoid seeing it abandoned after the Localization sub-items ;-) 
position

File: addons/website_customer/controllers/main.py [hunk 2]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
             domain += [
-                '|',
+                '|', '|',
                 ('name', 'ilike', post.get("search")),
-                ('website_description', 'ilike', post.get("search"))
+                ('website_description', 'ilike', post.get("search")),
+                ('industry_id.name', 'ilike', post.get("search")),
             ]
 
+        tag_id = post.get('tag_id', False)
```
Comment (odony) [inline addons/website_customer/controllers/main.py]: `dict.get()` returns `None` if the key is missing, so `post.get('tag_id')` is enough here.
Similarly, the declaration of `partner_name` above could have been simpler: `partner_name = post.get('search')` given how we use it afterwards (I know, that's not part of the PR ;))

File: addons/website_customer/controllers/main.py [hunk 3]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
 
         domain = [('website_published', '=', True), ('assigned_partner_id', '!=', False)]
         if partner_name:
             domain += [
-                '|',
+                '|', '|',
                 ('name', 'ilike', post.get("search")),
-                ('website_description', 'ilike', post.get("search"))
+                ('website_description', 'ilike', post.get("search")),
+                ('industry_id.name', 'ilike', post.get("search")),
```
Comment (odony) [inline addons/website_customer/controllers/main.py]: `post.get("search")` is `partner_name`  (probably deserves a better variable name, though)

File: addons/website_customer/controllers/main.py [hunk 4]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
             ]
 
+        tag_id = post.get('tag_id', False)
         if tag_id:
             tag_id = unslug(tag_id)[1] or 0
             domain += [('website_tag_ids', 'in', tag_id)]
 
+        # group by industry, based on customers found with the search(domain)
+        industries = Partner.sudo().read_group(domain, ["id", "industry_id"], groupby="industry_id", orderby="industry_id")
+        industry_count = Partner.sudo().search_count(domain)
```
Comment (odony) [inline addons/website_customer/controllers/main.py]: I know it's the same pattern as below, but `industry_count` is a confusing name for a variable holding a partner count

File: addons/website_customer/controllers/main.py [hunk 5]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
+
+        if industry:
+            domain += [('industry_id', '=', industry.id)]
+            if industry.id not in (x['industry_id'][0] for x in industries if x['industry_id']):
+                if industry.exists():
+                    industries.append({
+                        'industry_id_count': 0,
+                        'industry_id': (industry.id, industry.name)
+                    })
+                industries.sort(key=lambda d: d['industry_id'] and d['industry_id'][1])
```
Comment (odony) [inline addons/website_customer/controllers/main.py]: Should be indented further, re-sorting is only necessary if an entry was added?

File: odoo/addons/base/res/res_partner.py [hunk 6]
```diff
@@ -809,3 +810,12 @@ def _display_address_depends(self):
             'country_id.address_format', 'country_id.code', 'country_id.name',
             'company_name', 'state_id.code', 'state_id.name',
         ]
+
+
+class ResPartnerIndustry(models.Model):
+    _description = 'Sector of Activity'
+    _name = "res.partner.industry"
```
Comment (odony) [inline odoo/addons/base/res/res_partner.py]: `_order = 'name'`  otherwise they are sorted by ID

File: addons/website_customer/views/website_customer_templates.xml [hunk 7]
```diff
@@ -91,36 +94,53 @@
     </xpath>
 </template>
 
+<template id="opt_industry_list" inherit_id="website_customer.index" customize_show="True" name="Filter on Sector of Activity" priority="20">
```
Comment (odony) [inline addons/website_customer/views/website_customer_templates.xml]: :+1: for making the sector list an option!

File: addons/website_customer/views/website_customer_templates.xml [hunk 8]
```diff
@@ -201,6 +221,9 @@
                     <a class="media-heading" t-attf-href="/customers/#{slug(reference)}">
                         <span t-field="reference.self"/>
                     </a>
+                    <t t-if="partner.industry_id">
+                        <span class="label label-default"><t t-esc="partner.industry_id.name"/></span>
```
Comment (odony) [inline addons/website_customer/views/website_customer_templates.xml]: This displays the sector of activity of the Partner, not that of the Reference: `partner.industry_id` -> `reference.industry_id` (and in the `t-if`)

Comment (odony) [inline addons/website_customer/views/website_customer_templates.xml]: The Sector of activity of the Partner itself is missing in the details. It could be displayed next to its name as a badge, or in the left sidebar with the address.

File: addons/website_customer/controllers/main.py [hunk 9]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
+        '/customers/country/<model("res.country"):country>',
+        '/customers/country/<model("res.country"):country>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>',
+        '/customers/industry/<model("res.partner.industry"):industry>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>/page/<int:page>',
     ], type='http', auth="public", website=True)
-    def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
-        Country = request.env['res.country']
+    def customers(self, country=None, industry=None, page=0, **post):
```
Comment (odony) [inline addons/website_customer/controllers/main.py]: :+1: for pushing the tags into the query string. It will break a few old URLs, but such an explosion of routes would be unmaintainable.

File: addons/website_customer/controllers/main.py (hunk 5 above)
Comment (xmo-odoo) [inline addons/website_customer/controllers/main.py]: provide a "base value" if `d['industry_id']` is False/None, comparing different types is not supported anymore in Python 3.

File: addons/website_customer/controllers/main.py [hunk 10]
```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
...
         if tag_id:
             tag_id = unslug(tag_id)[1] or 0
             domain += [('website_tag_ids', 'in', tag_id)]
 
+        # group by industry, based on customers found with the search(domain)
+        industries = Partner.sudo().read_group(domain, ["id", "industry_id"], groupby="industry_id", orderby="industry_id")
+        industry_count = Partner.sudo().search_count(domain)
+
+        if industry:
+            domain += [('industry_id', '=', industry.id)]
```
Comment (xmo-odoo) [inline addons/website_customer/controllers/main.py]: use append

File: addons/website_customer/controllers/main.py [hunk 11]
```diff
@@ -85,8 +102,11 @@ def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
 
         values = {
             'countries': countries,
-            'current_country_id': country_id or 0,
-            'current_country': curr_country if country_id else False,
+            'current_country_id': country and country.id or 0,
```
Comment (xmo-odoo) [inline addons/website_customer/controllers/main.py]: Use if/else expression.

File: odoo/addons/base/security/ir.model.access.csv [hunk 12]
```diff
@@ -57,6 +57,8 @@
 "access_res_partner_bank_group_partner_manager","res_partner_bank group_partner_manager","model_res_partner_bank","group_partner_manager",1,1,1,1
 "access_res_partner_category_group_user","res_partner_category group_user","model_res_partner_category","group_user",1,0,0,0
 "access_res_partner_category_group_partner_manager","res_partner_category group_partner_manager","model_res_partner_category","group_partner_manager",1,1,1,1
+"access_res_partner_industry_group_user","res_partner_industry group_user","model_res_partner_industry","group_user",1,0,0,0
+"access_res_partner_industry_group_system","res_partner_industry group_system","model_res_partner_industry","group_system",1,1,1,1
```
Comment (nhomar) [inline odoo/addons/base/security/ir.model.access.csv]: it is common a sales manager do the add stuff foir this kind of clasification models, add the proper ACL I think.

Comment (odony) [inline odoo/addons/base/security/ir.model.access.csv]: Or not, as we're in `base` ;-)
